opentracing/tracecontext.go

```diff
@@ -43,11 +48,18 @@ type TraceContext interface {
 	// wish to use them as HTTP header keys (or key suffixes), and of course
 	// HTTP headers are case insensitive.
 	//
+	// `restrictedKey` MUST match the regular expression
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. See
+	// CanonicalizeTraceTagKey().
+	//
```

Comment:
I deliberately didn't specify the behavior here, though I would guess most "real" implementations wouldn't crash (since they don't *have* to).

Good call re the parameter, I will fix that.